README.md

```diff
@@ -19,6 +23,29 @@ A simple Ruby Gem to bootstrap dependencies for setting up and maintaining a loc
 
 *Note: You are not required to install Jekyll separately. Once the `github-pages` gem is installed, you can build your site using `jekyll build`, or preview your site using `jekyll serve`.* For more information about installing Jekyll locally, please see [the GitHub Help docs on the matter](https://help.github.com/articles/using-jekyll-with-pages#installing-jekyll).
 
+### Docker
+
+Provided that Docker is installed, one may avoid the setup of additional tools within the environment by simply spawning a Docker container.
+
+1. Run `make image` from the root of the pages-gem directory to build an image which will be tagged as `gh-pages`
+2. Start an instance of the server by running either:
+  - `SITE=PATH_TO_YOUR_PROJECT make server` from the root of the `gh-pages` repository (where the Makefile resides) or
+  - ``SITE=PATH_TO_YOUR_PROJECT docker run --rm -p 4000:4000 -v `realpath ${SITE}`:/site gh-pages`` from any directory or
+  - `github-pages $PATH_TO_YOUR_PROJECT` from any directory when [func.sh](func.sh) has been sourced into your terminal session or
+  - `github-pages` from the directory of the Jekyll site to be previewed when [func.sh](func.sh) has been sourced into your terminal session.
+
+**Note:** the `github-pages` function may be enabled by sourcing func.sh. This can be done by appending
+
+  ```bash
+  source $PATH_TO_THIS_DIRECTORY/func.sh
```

Comment:
This is great! Can it go in a `contrib/` subdirectory instead of the root of the project? `contrib/docker` maybe?

Reply:
Done :wink: f4a8d87bb0d74185b48d69d889e6940a683de132